### PR TITLE
Feature/AddsMyLocationAndUseZipCodeSwitches

### PR DESCRIPTION
**Pull request by cordhernandez**
#145
Works on Filter Delegate
Still In Progress

### CODE REVIEW

File: BlackNectar/BlackNectar/Colors.swift [hunk 1]
```diff
@@ -9,18 +9,115 @@
 import Foundation
 import UIKit
 
+//TODO: Replace this class with RedRomaColors to minimize Code Duplication
```
Comment (SirWellington) [inline BlackNectar/BlackNectar/Colors.swift:12]: Created #163 to handle this.

File: BlackNectar/BlackNectar/CustomLabelView.swift [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+    
+    func updateView() {
+        
+        layer.cornerRadius = cornerRadius
+        layer.borderWidth = borderWidth
+        layer.borderColor = borderColor.cgColor
+        layer.masksToBounds = true
+        
+        if shouldRasterize
+
```
Comment (SirWellington) [inline BlackNectar/BlackNectar/CustomLabelView.swift:87]: extra empty line here.

File: BlackNectar/BlackNectar/CustomLabelView.swift [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+    func updateView() {
+        
+        layer.cornerRadius = cornerRadius
+        layer.borderWidth = borderWidth
+        layer.borderColor = borderColor.cgColor
+        layer.masksToBounds = true
+        
+        if shouldRasterize
+            
+        {
```
Comment (SirWellington) [inline BlackNectar/BlackNectar/CustomLabelView.swift:88]: Braces should be consistent with others. Either on the same line, or the new line.

File: BlackNectar/BlackNectar/CustomThumbView.swift [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+//  BlackNectar
+//
+//  Created by Cordero Hernandez on 2/17/17.
+//  Copyright © 2017 BlackSource. All rights reserved.
+//
+
+import Foundation
+import UIKit
+
+class CustomThumbView: UIView {
```
Comment (SirWellington) [inline BlackNectar/BlackNectar/CustomThumbView.swift]: `@IBDesignable`

File: BlackNectar/BlackNectar/SearchStores.swift [hunk 5]
```diff
@@ -26,6 +26,15 @@ class SearchStores {
         
     }
     
+    static func searchForStoresByZipCode(withZipCode zipCode: String, callback: @escaping Callback) {
```
Comment (SirWellington) [inline BlackNectar/BlackNectar/SearchStores.swift]: Brilliant! Really clean code.

File: BlackNectar/BlackNectar/SearchStores.swift [hunk 6]
```diff
@@ -26,6 +26,15 @@ class SearchStores {
...
+        
+        let storesAPI = "https://blacknectar-api.blacksource.tech:9102/stores?zipCode=\(zipCode)"
+        guard let url = URL(string: storesAPI) else { return }
```
Comment (SirWellington) [inline BlackNectar/BlackNectar/SearchStores.swift]: This return should at least call the `callback` with no stores.

**Review by SirWellington — APPROVED** [review]
I'm seeing quite a bit of code duplication with the whole ZipCode/MyLocation thing. 

Maybe not in this ticket, but I suggest that we consolidate things so that we store a `CLLocationCoordinate2D` in the user preferences, and pass that around. 

This would work because both the `zipCode` and the `myLocation` eventually translate to a Latitude,Longitude.

File: BlackNectar/BlackNectar/Colors.swift (hunk 1 above)
Comment (cordhernandez) [reply BlackNectar/BlackNectar/Colors.swift:12]: ok

File: BlackNectar/BlackNectar/CustomLabelView.swift (hunk 3 above)
Comment (cordhernandez) [reply BlackNectar/BlackNectar/CustomLabelView.swift:88]: This has been fixed

File: BlackNectar/BlackNectar/CustomThumbView.swift (hunk 4 above)
Comment (cordhernandez) [reply BlackNectar/BlackNectar/CustomThumbView.swift]: done

File: BlackNectar/BlackNectar/SearchStores.swift (hunk 5 above)
Comment (cordhernandez) [reply BlackNectar/BlackNectar/SearchStores.swift]: 👊🏾

File: BlackNectar/BlackNectar/SearchStores.swift (hunk 6 above)
Comment (cordhernandez) [reply BlackNectar/BlackNectar/SearchStores.swift]: hmm what do you mean??

Comment (SirWellington) [reply BlackNectar/BlackNectar/SearchStores.swift]:
```swift

guard let url = URL(string: storesAPI) else {
    callback([])
    return
}
```

Comment (cordhernandez) [reply BlackNectar/BlackNectar/SearchStores.swift]: done

**Pull request merged**